controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -331,14 +332,14 @@ private SegmentRange convert(final String scope,
         return listOfSegment;
     }
 
-    public CompletableFuture<TxnStatus> commitTransaction(final String scope, final String stream, final TxnId
-            txnId) {
+    public CompletableFuture<TxnStatus> commitTransaction(final String scope, final String stream, final TxnId txnId, 
+                                                          final String writerId, final long timestamp) {
```

Comment:
This is not on grpc. This is an internal method. 
Grpc contract is both forward and backward compatible because we add new fields to the rpc api contract.